File: src/main/java/com/example/productorderservice/order/Order.java
package com.example.productorderservice.order;

import com.example.productorderservice.product.Product;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "orders")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
class Order {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne
    private Product product;
    private int quantity;

    public Order(Product product, int quantity) {

        this.product = product;
        this.quantity = quantity;
    }
}

File: test_cafekiosk/src/main/kotlin/com/testcafekiosk/spring/domain/order/Order.kt
package com.testcafekiosk.spring.domain.order

import com.testcafekiosk.spring.domain.BaseEntity
import com.testcafekiosk.spring.domain.product.Product
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "orders")
class Order(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Enumerated(EnumType.STRING)
    val orderStatus: OrderStatus,

    val totalPrice: Int,

    val registeredDateTime: LocalDateTime,

    @OneToMany(mappedBy = "order", cascade = [CascadeType.ALL])
    var orderProducts: MutableList<OrderProduct> = mutableListOf()

) : BaseEntity() {

    fun registerOrderProduct(products: List<Product?>) {
        val orderProducts = products
            .map {
                OrderProduct(
                    order = this,
                    product = it!!
                )
            }
            .toMutableList()
        this.orderProducts = orderProducts
    }

    companion object {
        fun from(
            products: List<Product?>,
            registeredDateTime: LocalDateTime,
            orderStatus: OrderStatus = OrderStatus.INIT
        ): Order {
            val order = Order(
                orderStatus = orderStatus,
                totalPrice = products.sumOf { it!!.price },
                registeredDateTime = registeredDateTime,
//                orderProducts = registerOrderProduct(products)
            )
            order.registerOrderProduct(products)
            return order
        }
    }


}